### PR TITLE
API changes to support robot description

### PR DESCRIPTION
**Pull request by destogl**
As discussed in #66. This replaces it.

The Admittance Controller changes are in ros-controls/ros2_controllers#

### CODE REVIEW

File: kinematics_interface_kdl/src/kinematics_interface_kdl.cpp [hunk 1]
```diff
@@ -21,36 +21,57 @@ namespace kinematics_interface_kdl
...
+  {
+    parameters_interface->get_parameter(ns + "tip", end_effector_name);
+  }
+
   // create kinematic chain
   KDL::Tree robot_tree;
-  kdl_parser::treeFromString(robot_description, robot_tree);
-
+  kdl_parser::treeFromString(robot_description_local, robot_tree);
+
```
Comment (destogl) [inline kinematics_interface_kdl/src/kinematics_interface_kdl.cpp:69]:
```suggestion

```

File: kinematics_interface/include/kinematics_interface/kinematics_interface.hpp [hunk 2]
```diff
@@ -39,10 +39,15 @@ class KinematicsInterface
...
+   * \param[in] robot_description robot URDF in string format
+   * \param[in] parameters_interface 
+   * \param[in] param_namespace namespace for kinematics parameters - defaults to "kinematics"
+   * \return true if successful
    */
   virtual bool initialize(
+    const std::string & robot_description,
     std::shared_ptr<rclcpp::node_interfaces::NodeParametersInterface> parameters_interface,
-    const std::string & end_effector_name) = 0;
+    const std::string & param_namespace = "kinematics") = 0;
```
Comment (saikishor) [inline kinematics_interface/include/kinematics_interface/kinematics_interface.hpp:50]: Do we want to hardcode this here?

File: kinematics_interface_kdl/src/kinematics_interface_kdl.cpp [hunk 3]
```diff
@@ -21,36 +21,56 @@ namespace kinematics_interface_kdl
...
-  if (parameters_interface->has_parameter("alpha"))
+  if (parameters_interface->has_parameter(ns + "alpha"))
   {
-    parameters_interface->get_parameter("alpha", alpha_param);
+    parameters_interface->get_parameter(ns + "alpha", alpha_param);
   }
   alpha = alpha_param.as_double();
+  // get end-effector name
+  auto end_effector_name = rclcpp::Parameter("tip", "MISSING_END_EFFECTOR_NAME");
+  if (parameters_interface->has_parameter(param_namespace + "tip"))
```
Comment (saikishor) [inline kinematics_interface_kdl/src/kinematics_interface_kdl.cpp:61]:
```suggestion
  if (parameters_interface->has_parameter(ns + "tip"))
```

File: kinematics_interface_kdl/src/kinematics_interface_kdl.cpp [hunk 4]
```diff
@@ -21,36 +21,56 @@ namespace kinematics_interface_kdl
...
   auto alpha_param = rclcpp::Parameter("alpha", 0.000005);
-  if (parameters_interface->has_parameter("alpha"))
+  if (parameters_interface->has_parameter(ns + "alpha"))
   {
-    parameters_interface->get_parameter("alpha", alpha_param);
+    parameters_interface->get_parameter(ns + "alpha", alpha_param);
   }
   alpha = alpha_param.as_double();
+  // get end-effector name
+  auto end_effector_name = rclcpp::Parameter("tip", "MISSING_END_EFFECTOR_NAME");
```
Comment (saikishor) [inline kinematics_interface_kdl/src/kinematics_interface_kdl.cpp:60]:
```suggestion
  auto end_effector_name = rclcpp::Parameter();
```

Do we need this?
why not add check after the get_parameter and if it is empty, then print error and return false?

`if (!robot_tree.getChain(root_name_, end_effector_name, chain_))`

Here in L81 it expects a string I believe, I'm not sure though. If is it a string, then it needs to be changed

**Review by saikishor — CHANGES_REQUESTED** [review]
Minor comments
The rest looks great :)
Thanks for the nice work

File: kinematics_interface/include/kinematics_interface/kinematics_interface.hpp (hunk 2 above)
Comment (destogl) [reply kinematics_interface/include/kinematics_interface/kinematics_interface.hpp:50]: You are right, we don't need this here:

```suggestion
    const std::string & param_namespace) = 0;
```

File: kinematics_interface_kdl/include/kinematics_interface_kdl/kinematics_interface_kdl.hpp [hunk 5]
```diff
@@ -40,8 +40,9 @@ class KinematicsInterfaceKDL : public kinematics_interface::KinematicsInterface
 {
 public:
   bool initialize(
+    const std::string & robot_description,
     std::shared_ptr<rclcpp::node_interfaces::NodeParametersInterface> parameters_interface,
-    const std::string & end_effector_name) override;
+    const std::string & param_namespace = "kinematics") override;
```
Comment (destogl) [inline kinematics_interface_kdl/include/kinematics_interface_kdl/kinematics_interface_kdl.hpp:45]:
```suggestion
    const std::string & param_namespace) override;
```

File: kinematics_interface_kdl/src/kinematics_interface_kdl.cpp (hunk 4 above)
Comment (Nibanovic) [reply kinematics_interface_kdl/src/kinematics_interface_kdl.cpp]: We don't, thanks!

Now returning `false` on missing `tip` parameter. Added a small test for this as well.

File: kinematics_interface/include/kinematics_interface/kinematics_interface.hpp [hunk 6]
```diff
@@ -39,10 +39,15 @@ class KinematicsInterface
...
+   * \param[in] robot_description robot URDF in string format
+   * \param[in] parameters_interface 
+   * \param[in] param_namespace namespace for kinematics parameters - defaults to "kinematics"
+   * \return true if successful
    */
   virtual bool initialize(
+    const std::string & robot_description,
     std::shared_ptr<rclcpp::node_interfaces::NodeParametersInterface> parameters_interface,
-    const std::string & end_effector_name) = 0;
+    const std::string & param_namespace) = 0;
```
Comment (bmagyar) [inline kinematics_interface/include/kinematics_interface/kinematics_interface.hpp:50]: if we only introduced a new initialize argument we could put this in humble or that's not an interest?

**Review by bmagyar — CHANGES_REQUESTED** [review]

**Review by bmagyar — APPROVED** [review]

**Pull request merged**